src/vmm/src/builder.rs

```diff
@@ -520,6 +560,12 @@ pub fn build_microvm_from_snapshot(
             .clone(),
     )?;
 
+    // Inject the notification to VMGenID that we have resumed from a snapshot.
+    // This needs to happen before we resume vCPUs, so that we minimize the time between vCPUs
+    // resuming and notification being handled by the driver.
+    #[cfg(target_arch = "x86_64")]
+    vmm.acpi_device_manager.notify_vmgenid()?;
```

Comment:
1. Comment still here
2. If this does not need to be after `start_vcpus` can you move up to the acpi section just before `start_vcpus`?

Reply:
The comment about injecting the notification before resuming the vCPUs is valid. Before or after the `start_vcpus` doesn't make any difference, so why move it?

Reply:
Sorry, did not read comment carefully enough.
The move is mainly to combine 2 `#[cfg(target_arch = "x86_64")]` sections into 1 that deals with all acpi related things.

Reply:
Ok, that makes sense. Moved it.